### PR TITLE
Basic refactor for MediaType

### PR DESCRIPTION
**Pull request by jsancio**
Finish this tasks before merging:

- [x] Analyze the dependency report and make sure that we are using the same version and there are any major evictions.

### CODE REVIEW

File: cosmos-server/src/main/scala/com/mesosphere/cosmos/repository/UniverseClient.scala [hunk 1]
```diff
@@ -383,16 +379,6 @@ object DefaultUniverseClient {
...
     List(MediaTypes.UniverseV4Repository, MediaTypes.UniverseV3Repository, MediaTypes.UniverseV2Repository)
-
-  def parseContentType(header: Option[String]): TwitterTry[MediaType] = {
-    TwitterTry(header.getOrElse(throw UnsupportedContentType(SupportedMediaTypes).exception))
-      .flatMap(MediaTypeParser.parse)
-      .handle {
-        case MediaTypeParseError(_, _) =>
-          throw UnsupportedContentType(SupportedMediaTypes).exception
-      }
-  }
```
Comment (jsancio) [inline cosmos-server/src/main/scala/com/mesosphere/cosmos/repository/UniverseClient.scala]: This is not used anywhere. Is this a bug or it is okay to delete this? cc @takirala

File: cosmos-common/src/main/scala/com/mesosphere/http/QualityValue.scala [hunk 2]
```diff
@@ -19,15 +21,15 @@ object QualityValue {
   def parse(value: String): Try[QualityValue] = {
```
Comment (jsancio) [inline cosmos-common/src/main/scala/com/mesosphere/http/QualityValue.scala]: This is only used by tests. Should we just remove this? cc @takirala @larioj

**Comment by jsancio** [comment]
@larioj This is ready for review!

File: cosmos-common/src/main/scala/com/mesosphere/http/QualityValue.scala (hunk 2 above)
Comment (larioj) [reply cosmos-common/src/main/scala/com/mesosphere/http/QualityValue.scala]: That's fine by me

**Comment by jsancio** [comment]
Thanks for the review @larioj !

**Pull request merged**